bno055/connectors/i2c.py

```diff
@@ -27,22 +27,47 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
 
-from bno055.connectors.Connector import Connector
+from smbus import SMBus
+
 from rclpy.node import Node
 
+from bno055 import registers
+from bno055.connectors.Connector import Connector
+from bno055.error_handling.exceptions import TransmissionException
+
 
 class I2C(Connector):
     CONNECTIONTYPE_I2C = 'i2c'
 
-    def __init__(self, node: Node):
-        # Initialize parent
+    def __init__(self, node: Node, i2c_bus=0, i2c_addr=registers.BNO055_ADDRESS_A):
         super().__init__(node)
+        self.bus = SMBus(i2c_bus)
+        self.address = i2c_addr
 
     def connect(self):
-        raise NotImplementedError('I2C not yet implemented')
+        returned_id = self.bus.read_byte_data(self.address, registers.BNO055_CHIP_ID_ADDR)
+        if returned_id != registers.BNO055_ID:
+            raise TransmissionException('I2C not yet implemented')
```

Comment:
@asymingt should this instead be an exception like:

```suggestion
            raise TransmissionException('Could not get BNO055 chip ID via I2C')
```